### ISSUE TITLE
Feature Request: Add Edit Distance metric_fn to tf.metrics or Make a way to Pass Custom metric_fn outside tf.metrics

### ISSUE THREAD

**Issue by selcouthlyBlue** [tensorflow/tensorflow#16999]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 8.1
- **TensorFlow installed from (source or binary)**:
Binary
- **TensorFlow version (use command below)**:
1.4 CPU
- **Python version**: 
3.6
- **CUDA/cuDNN version**:
N/A
- **GPU model and memory**:
N/A

### Describe the problem
I would like another metric (edit_distance) to be added into tf.metrics for `label_error_rate` calculation. I'm having trouble passing metric dicts for `eval_metric_ops` to be used by `ModelFnOps` that are not part of  tf.metrics. 

If that isn't possible, would it be possible for us we can pass our own `metric_fn`s that are outside tf.metrics? 

### Source code / logs
This is the metric dict I would like to pass as the `eval_metric_ops` for `ModelFnOps`:

`metric = {"label_error_rate": tf.reduce_mean(tf.edit_distance(tf.cast(y_pred, tf.int32), y_true), name="label_error_rate")}`

Where `y_pred` and `y_true` are sparse.

**Comment by bignamehyp** [comment]
We will consider it if we heard more demands for this request.

**Comment by selcouthlyBlue** [comment]
Alright

**Comment by ngs2410** [comment]
I needed this today too.

**Comment by selcouthlyBlue** [comment]
One workaround I made is wrapped the `edit_distance` metric with `tf.metrics.mean` like so `edit_distance = tf.metrics.mean(tf.reduce_mean(tf.edit_distance(tf.cast(y_pred, tf.int32), y_true), name=name))` and passed that as the dict value for `eval_metric_ops` `{"edit_distance": edit_distance}`. It works but I'm not sure if this is the proper way of doing it.

**Comment by tatianashp** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
